hw1/app.js

```diff
@@ -0,0 +1,334 @@
+    }
+
+    render () {
+        // using innerHTML to render inventory listing
+    }
+}
+
+class Menu {
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        this.init();
+        this.food= this.store.foods;
+    }
+
+    init () {
+        this.render();
+    }
+
+    render () {
+        // TODO: render a list of food menu from store using innerHTML
+        var foods = this.store.foods;
+        for(var i=0; i< foods.length; i ++){
+            console.log(foods[i]);
+        }
+
+
+        let tbody = this.root.querySelector('tbody');
+        tbody.innerHTML = this.getHTMLForCart(foods);
+
+        let deleteBtn = this.root.querySelectorAll('.deletebutton');
+        let listfood = this.store.foods;
+        for(var i=0; i <deletebutton.length; i++){
+            let deleteBtn = deletebutton[i];
+            let Index = i;
+            deleteBtn.addEventListener('click',( )=>{
+                listfood.splice(Index,1);
+                this.store.foods = listfood;
+                alert("delete item");
+                this.render();
+
+        
+        //tbody.innerHTML = this.addItemtoCart();
+
+        //let deletebutton = this.root.querySelectorAll('.deletebutton');
+        
+        
+        //for(var i = 0; i < deletebutton.length; i ++){
+          //  let deleteBtn = deletebutton[i];
+           // let deletelist = this.store.cartItems;
+           // let index = i;
+           // deleteBtn.addEventListener('click',( )=>{
+                
+               // deletelist.splice(index,1);
+                //this.store.cartItems = deletelist;
+               // alert("delete item");
+                //this.render();
+            });
+        }
+    }
+
+    getHTMLForCart(food){
+        var resuelt ='';
+        for(var i =0; i <food.length; i ++){
+            result = result + "<tr><td>" + food[i].name
+                            + "</td><td>" + food[i].price
+                            + "</td><td> <button class = 'deletebutton'>Delete</button></td></tr>"  
+        }
+        return result;
+    }
+}
+
+class CreateFood {
+    constructor(root, store) {
+        this.root = root; // root should be the container of the form itself
+        this.store = store;
+        this.init();
+    }
+
+    init () {
+        // attach click event to create button
+        this.root.addEventListener('click', this.onClick);
+    }
+    destory(){
+        this.root.addEventListener('click',this.onClick);
+    }
+
+    createFood () {
+        // will need to do querySelector to find out every single form element
+        // to get their values before creating a new food
+        // after creating a new food item, add it to store
+    let foods = this.store.foods || [];
+    console.log(this.root.dataset);
+    foors.push({
+
+        name:this.root.dataset.name,
+        price:this.root.dataset.price
+    });
+    console.log(foods);
+    this.store.foods = foods;
```

Comment:
Create food should be taking the input value than the dataset attribute. In example, you will need to do a query selector to find out the input DOM node from the html in the create-food.html first. Then, you can get its value by calling `.value` property.